superset/security/manager.py

```diff
@@ -1872,14 +1873,35 @@ def raise_for_access(
                         .one_or_none()
                     )
                     and dashboard_.roles
-                    and (slice_id := form_data.get("slice_id"))
                     and (
-                        slc := self.get_session.query(Slice)
-                        .filter(Slice.id == slice_id)
-                        .one_or_none()
+                        (
```

Comment:
```suggestion
                            (
```

Reply:
maybe @Vitor-Avila or @jfrag1 can help add if there's anything else we need to do for embedded with the below logic? Check for a guest user?

Reply:
Yes I will add logic for embedded in a separate PR

Reply:
@eschutho Black formats the code how it sees fit.